src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetAzurePrimaryDataStep.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.dataset.flight.delete;
+
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.azure.blobstore.AzureBlobStorePdao;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+public class DeleteDatasetAzurePrimaryDataStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteDatasetAzurePrimaryDataStep.class);
+
+  private final AzureBlobStorePdao azureBlobStorePdao;
+  private final FireStoreDao fileDao;
+  private final DatasetService datasetService;
+  private final UUID datasetId;
+  private final ConfigurationService configService;
+
+  public DeleteDatasetAzurePrimaryDataStep(
+      AzureBlobStorePdao azureBlobStorePdao,
+      FireStoreDao fileDao,
+      DatasetService datasetService,
+      UUID datasetId,
+      ConfigurationService configService) {
+    this.azureBlobStorePdao = azureBlobStorePdao;
+    this.fileDao = fileDao;
+    this.datasetService = datasetService;
+    this.datasetId = datasetId;
+    this.configService = configService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+    if (configService.testInsertFault(ConfigEnum.LOAD_SKIP_FILE_LOAD)) {
+      // If we didn't load files, don't try to delete them
+      fileDao.deleteFilesFromDataset(dataset, fireStoreFile -> {});
+    } else {
+      fileDao.deleteFilesFromDataset(dataset, azureBlobStorePdao::deleteFile);
+    }
+
+    // this fault is used by the DatasetConnectedTest > testOverlappingDeletes
+    if (configService.testInsertFault(ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT)) {
+      logger.info("DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT");
+      while (!configService.testInsertFault(
+          ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT)) {
+        logger.info("Sleeping for CONTINUE FAULT");
+        TimeUnit.SECONDS.sleep(5);
+      }
+      logger.info("DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT");
+    }
+
+    FlightMap map = context.getWorkingMap();
+    map.put(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.NO_CONTENT);
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) {
```

Comment:
Not a huge deal, but it looks like we have a ton of steps that have an undo that always returns success. These could all have a common subclass to avoid the duplication, e.g. `abstract class NoUndoStep implements Step { ... }`.

Reply:
Yeah, I like that.  What about even more broad (e.g. `DefaultStepImpl`) that would do that and any other common patterns (just cause we can't extend multiple classes)

Reply:
I didn't see other cases where a subclass would help, and I think we have `FlightUtils` for other common step code. (Also I'm not a fan of names like `Impl` since they have no meaning; although sometimes there's no better name to use...)